components/webext-storage/src/sync/bridge.rs

```diff
@@ -20,14 +21,21 @@ const SYNC_ID_META_KEY: &str = "sync_id";
 /// `storage.sync` store work with Desktop's Sync implementation.
 /// Conceptually, it's similar to `sync15::Store`, which we
 /// should eventually rename and unify with this trait (#2841).
-pub struct BridgedEngine<'a> {
-    db: &'a StorageDb,
+///
+/// Unlike most of our other implementation which hold a strong reference
+/// to the store, this engine keeps a weak reference in an attempt to keep
+/// the desktop semantics as close as possible to what they were when the
+/// engines all took lifetime params to ensure they don't outlive the store.
```

Comment:
In other words: What used to be enforced by the compiler, will now become an error when we upgrade the reference. 

I think it's totally fine, and worth it to make the engine Sync + Send to be UniFFIed!

Reply:
Yep! This "one strong ref, many weak refs" pattern also helps when `close()` is racing against other operations. In that case, we don't want those other operations to prevent the connection from closing.